src/Entities/Casts/Related/Zaakinformatieobjecten.php

```diff
@@ -4,20 +4,29 @@
 
 namespace OWC\Zaaksysteem\Entities\Casts\Related;
 
-use OWC\Zaaksysteem\Entities\Entity;
-use OWC\Zaaksysteem\Support\Collection;
 use OWC\Zaaksysteem\Endpoints\Filter\ZaakinformatieobjectenFilter;
-
+use OWC\Zaaksysteem\Entities\Enkelvoudiginformatieobject;
+use OWC\Zaaksysteem\Entities\Entity;
+use OWC\Zaaksysteem\Entities\Zaakinformatieobject;
 use function OWC\Zaaksysteem\Foundation\Helpers\resolve;
 
+use OWC\Zaaksysteem\Support\Collection;
+
 class Zaakinformatieobjecten extends ResourceCollection
 {
     public function resolveRelatedResourceCollection(Entity $entity): Collection
     {
         $statussenEndpoint = resolve($this->clientName)->zaakinformatieobjecten();
-
         $filter = new ZaakinformatieobjectenFilter();
 
-        return $statussenEndpoint->filter($filter->byZaak($entity));
+        $objects = $statussenEndpoint->filter($filter->byZaak($entity));
+
+        return $objects->filter(function ($object) {
+            if (! $object instanceof Zaakinformatieobject || ! $object->informatieobject instanceof Enkelvoudiginformatieobject) {
```

Comment:
Klopt dit wel? Door de `||` zal dit altijd `false` returnen als `$object` geen `Zaakinformatieobject` is, toch?

Reply:
Nee, wellicht overheen gelezen hoor. Moest zelf ook 2 keer kijken.
De eerste kijkt of het `$object` een instantie is van Zaakinformatieobject.
De tweede kijkt of het informatieobject in `$object` een instantie is van Enkelvoudiginformatieobject.

Er waren gevallen waarin `$object` een instantie was van Zaakinformatieobject maar dat $object->informatieobject een `null` was. Waardoor je dit $object dan niet nodig hebt.

Reply:
Oh ik zie nu wat je bedoelt. Heb echt helemaal verkeerd gelezen zie ik nu 😂 Thanks voor de toelichting